Support customization of link text

yanis-sadeg opened the pull request:
## Description
Added functionality to add text to the location button. 
It is activated with the "text" option and does not allow HTML characters.

Tested on Firefox, Chrome, Safari. ✔️

## Example
```
lc = L.control.locate({
    strings: {
        title: "Show me where I am, yo!",
        text: "Locate me"
    }
}).addTo(map);
```

## Screenshot
![Leaflet locate link text](https://i.imgur.com/SnNWSId.png)

Commit:
Support customization of link text

yanis-sadeg committed:
Update README.md

The link of the demo now redirects correctly in https.

yanis-sadeg committed:
Update L.Control.Locate.js

Update of the formatting.

domoritz commented on README.md:
```diff
@@ -213,7 +213,7 @@ var map = new L.Map('map', {
 ## Developers
 
-Run the demo locally with `grunt serve` and then open [localhost:9000/demo/index.html](http://localhost:9000/demo/index.html).
+Run the demo locally with `grunt connect` and then open [localhost:9000/demo/index.html](http://localhost:9000/demo/index.html).
```
```suggestion
Run the demo locally with `yarn start` or `npm run start` and then open [localhost:9000/demo/index.html](http://localhost:9000/demo/index.html).
```

domoritz commented on src/L.Control.Locate.js:
```diff
@@ -283,6 +285,14 @@ You can find the project at: https://github.com/domoritz/leaflet-locatecontrol
                 link.href = '#';
                 var icon = L.DomUtil.create(options.iconElementTag, options.icon, link);
+
+                if(options.strings.text !== undefined) {
```
```suggestion
                if (options.strings.text !== undefined) {
```

domoritz commented on src/L.Control.Locate.js:
```diff
@@ -283,6 +285,14 @@ You can find the project at: https://github.com/domoritz/leaflet-locatecontrol
+
+                if(options.strings.text !== undefined) {
+                    var text = L.DomUtil.create(options.textElementTag, 'leaflet-locate-text', link);
+                    text.textContent = '\xa0' + options.strings.text;
```
Why do you need to add `\xa0`?

yanis-sadeg replied:
> 
> 
> Why do you need to add `\xa0`?

To make a space between the icon and the text, otherwise they are stuck together and it's ugly. I had the choice to do it by css or like this, i choosen this method. But thinking about, maybe I should make the space by css on the icon with a padding right, like this if the developer choose to use only the text without the icon there is no space for nothing. What do you think?

domoritz replied:
Yep, let's do it in CSS.

domoritz commented on src/L.Control.Locate.js:
```diff
@@ -283,6 +285,14 @@ You can find the project at: https://github.com/domoritz/leaflet-locatecontrol
+                if(options.strings.text !== undefined) {
+                    var text = L.DomUtil.create(options.textElementTag, 'leaflet-locate-text', link);
+                    text.textContent = '\xa0' + options.strings.text;
+		    link.classList.add('leaflet-locate-text-active');
```
indentation?

yanis-sadeg replied:
> 
> 
> indentation?

Yes... In my side it's correctly indented, I don't know why the indentation on this preview is wrong. I will check about it for the next push.

domoritz replied:
Make sure you use spaces.

domoritz commented on src/L.Control.Locate.scss:
```diff
@@ -15,6 +15,15 @@
   }
 }
 
+.leaflet-touch .leaflet-bar .leaflet-locate-text-active  {
```
What does this have to do with the proposed change?

yanis-sadeg replied:
This allows to modify the css only if the text option is used, it will not impact the button if the option is not used. Also, I don't use the .leaflet-control-locate because otherwise I would have to use !important on the width property.

Modifications have been made on the css if the text option is displayed because of the default properties of leaflet which made it impossible to display the text correctly.

domoritz reviewed (approved):
Thanks for the pull request. Can you add a screenshot to the description?

yanis-sadeg commented:
> 
> 
> Thanks for the pull request. Can you add a screenshot to the description?

Sure, I will on the next push.

Edite: The requested changes have been made.

yanis-sadeg committed:
Update README.md

The way to start the project has been changed.

Co-authored-by: Dominik Moritz <domoritz@gmail.com>

yanis-sadeg committed:
Update src/L.Control.Locate.js

formatting changed

Co-authored-by: Dominik Moritz <domoritz@gmail.com>

Commit:
Added screenshot-text.png

README.md updated: The new screenshot-text.png has been added

A padding right has been added on the icon's selector (works only if the text option is activated to avoid an extra padding)

Commit:
Icon padding works only if the text and icon are used together

Commit:
Unused variable removed

Commit:
indentation

domoritz commented:
I meant a screenshot in the pull request description not the code. I just want some documentation for the feature here. Sorry for being ambiguous.

Commit:
Screenshot-text.png removed

yanis-sadeg commented:
> 
> 
> I meant a screenshot in the pull request description not the code. I just want some documentation for the feature here. Sorry for being ambiguous.

My bad, I had misunderstood. Done 👌

domoritz commented on src/L.Control.Locate.js:
```diff
@@ -283,6 +285,17 @@ You can find the project at: https://github.com/domoritz/leaflet-locatecontrol
+                    text.textContent = options.strings.text;
+		            link.classList.add('leaflet-locate-text-active');
+                    link.parentNode.style.display = "flex";
+                    if(options.icon.length > 0) {
```
```suggestion
                    if (options.icon.length > 0) {
```

domoritz committed:
Update src/L.Control.Locate.js

domoritz commented:
Thank you for the addition!
